libs/ledger-live-common/src/families/aptos/js-getFeesForTransaction.ts

```diff
@@ -69,7 +69,14 @@ export const getFee = async (
             break;
           }
           default: {
-            throw Error(`Simulation failed with following error: ${completedTx.vm_status}`);
+            // HOTFIX: we need to check if there is enough balance for gas
+            // idealy it should be covered with EINSUFFICIENT_BALANCE from network
+            const expectedGas = BigNumber(gasLimit * gasPrice);
+            if (transaction.amount.plus(expectedGas).isGreaterThan(account.spendableBalance)) {
+              break;
```

Comment:
Is it possible to move this check to an early exit before line 54 `const simulation = await aptosClient.simulateTransaction(publicKeyEd, tx);`?
It seems like it to me, and that would result in tidier code (not a fan of test inside default cases)